master/internal/resourcemanagers/job.go

```diff
@@ -69,102 +70,57 @@ func assignmentIsScheduled(allocatedResources *sproto.ResourcesAllocated) bool {
-	// 			groups[reqs[i].Group].qPosition = prev + 1
-	// 		} else {
-	// 			// seek the next populated value
-	// 			var seeker int
-	// 			for loc := i + 1; loc < len(reqs); loc++ {
-	// 				seeker = loc
-	// 				if groups[reqs[loc].Group].qPosition != -1 {
-	// 					break
-	// 				} else if loc == len(reqs) {
-	// 					break
-	// 				}
-	// 			}
-
-	// 			// set queue positions
-	// 			if seeker >= len(reqs)-1 {
-	// 				for setter := i; setter < len(reqs); setter++ {
-	// 					groups[reqs[setter].Group].qPosition = prev + 1
-	// 					prev++
-	// 				}
-	// 			} else {
-	// 				maxValue := groups[reqs[seeker].Group].qPosition
-	// 				diff := float64(seeker - i)
-	// 				increment := (maxValue - prev) / diff
-
-	// 				for setter := i; setter < seeker; setter++ {
-	// 					groups[reqs[setter].Group].qPosition = prev + increment
-	// 					prev += increment
-	// 				}
-	// 			} // find the value of the non negative position and add increments
-	// 		}
-	// 	}
-	// 	prev = groups[reqs[i].Group].qPosition
-	// 	i++
-	// }
-}
-
 // could be swapped for another job order representation
 // 0 or nonexisting keys mean that it needs to initialize.
-type jobSortState = map[model.JobID]float64
+type jobSortState = map[model.JobID]decimal.Decimal
 
 func initalizeJobSortState() jobSortState {
 	state := make(jobSortState)
-	state[job.HeadAnchor] = 0
+	state[job.HeadAnchor] = decimal.New(0, job.DecimalExp)
 	state[job.TailAnchor] = initalizeQueuePosition(time.Now())
 	return state
 }
 
-func computeNewJobPos(msg job.MoveJob, qPositions jobSortState) (float64, bool, error) {
+func computeNewJobPos(msg job.MoveJob, qPositions jobSortState) (decimal.Decimal, error) {
 	if msg.Anchor1 == msg.ID || msg.Anchor2 == msg.ID {
-		return 0, false, fmt.Errorf("cannot move job relative to itself")
+		return decimal.NewFromInt(0), fmt.Errorf("cannot move job relative to itself")
 	}
 
 	qPos1, ok := qPositions[msg.Anchor1]
 	if !ok {
-		return 0, false, fmt.Errorf("could not find anchor job %s", msg.Anchor1)
+		return decimal.NewFromInt(0), fmt.Errorf("could not find anchor job %s", msg.Anchor1)
 	}
 
 	qPos2, ok := qPositions[msg.Anchor2]
 	if !ok {
-		return 0, false, fmt.Errorf("could not find anchor job %s", msg.Anchor2)
+		return decimal.NewFromInt(0), fmt.Errorf("could not find anchor job %s", msg.Anchor2)
 	}
 
 	qPos3, ok := qPositions[msg.ID]
 	if !ok {
-		return 0, false, fmt.Errorf("could not find job %s", msg.ID)
+		return decimal.NewFromInt(0), fmt.Errorf("could not find job %s", msg.ID)
 	}
 
 	// check if qPos3 is between qPos1 and qPos2
-	smallPos := math.Min(qPos1, qPos2)
-	bigPos := math.Max(qPos1, qPos2)
-	if qPos3 > smallPos && qPos3 < bigPos {
-		return 0, false, nil // no op. Job is already in the correct position.
+	smallPos := decimal.Min(qPos1, qPos2)
+	bigPos := decimal.Max(qPos1, qPos2)
+	if qPos3.GreaterThan(smallPos) && qPos3.LessThan(bigPos) {
+		return decimal.NewFromInt(0), nil // no op. Job is already in the correct position.
```

Comment:
shouldn't this return `qPos3` back. if there is no error it looks like we set it to pos `decimal.NewFromInt(0)`

Reply:
ah great catch